subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c

```diff
@@ -25,11 +25,17 @@ LOG_MODULE_DECLARE(mcumgr_img_grp, CONFIG_MCUMGR_GRP_IMG_LOG_LEVEL);
 #define SLOT1_PARTITION		slot1_partition
 #define SLOT2_PARTITION		slot2_partition
 #define SLOT3_PARTITION		slot3_partition
+#define SLOT4_PARTITION		slot4_partition
+#define SLOT5_PARTITION		slot5_partition
 
 BUILD_ASSERT(CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 1 ||
-	     (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 2 &&
+	     (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER >= 2 &&
 	      FIXED_PARTITION_EXISTS(SLOT2_PARTITION) &&
-	      FIXED_PARTITION_EXISTS(SLOT3_PARTITION)),
+	      FIXED_PARTITION_EXISTS(SLOT3_PARTITION)) ||
+	     (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 3 &&
```

Comment:
Actually the check would not work as expected for `CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 3` (for `3`, the build assert would return earlier without checking the last condition). Please consider my previous suggestion as fix for this one.